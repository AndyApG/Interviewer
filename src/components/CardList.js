import React from 'react';
import { compose } from 'redux';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { scrollable, selectable } from '../behaviours';
import { Card } from '.';
import { entityPrimaryKeyProperty } from '../ducks/modules/network';

const EnhancedCard = selectable(Card);

/**
  * Card List
  */
const CardList = (props) => {
  const {
    className,
    details,
    label,
    items,
    onItemClick,
    isItemSelected,
    getKey,
  } = props;

  const classNames = cx('card-list', className);

  return (
    <div className={classNames}>
      {
        items.map(node => (
          <span className="card-list__content" key={getKey(node)}>
            <EnhancedCard
              label={label(node)}
              selected={isItemSelected(node)}
              details={details(node)}
              onSelected={() => onItemClick(node)}
            />
          </span>
        ))
      }
    </div>
  );
};

CardList.propTypes = {
  className: PropTypes.string,
  details: PropTypes.func,
  label: PropTypes.func,
  items: PropTypes.array.isRequired,
  onItemClick: PropTypes.func,
  isItemSelected: PropTypes.func,
  getKey: PropTypes.func,
};

CardList.defaultProps = {
  className: '',
  details: () => (''),
  label: () => (''),
  items: [],
  onItemClick: () => {},
  isItemSelected: () => false,
  getKey: node => node[entityPrimaryKeyProperty],
};

export default compose(
  scrollable,
)(CardList);
