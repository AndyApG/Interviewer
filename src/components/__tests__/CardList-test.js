/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import CardList from '../CardList';

const nodes = [
  { uid: 'a', name: 'a name', age: '22' },
  { uid: 'b', name: 'b name', age: '88' },
  { uid: 'c', name: 'c name', age: '33' },
];

describe('CardList component', () => {
  it('renders cards with list', () => {
    const component = shallow(
      <CardList nodes={nodes} label={node => node.name} details={node => [{ age: `${node.age}` }]} />,
    );

    expect(component).toMatchSnapshot();
  });
});
