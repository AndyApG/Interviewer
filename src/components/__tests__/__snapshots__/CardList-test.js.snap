// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component renders cards with list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scrollable
    details={[Function]}
    label={[Function]}
    nodes={
        Array [
            Object {
              "age": "22",
              "name": "a name",
              "uid": "a",
            },
            Object {
              "age": "88",
              "name": "b name",
              "uid": "b",
            },
            Object {
              "age": "33",
              "name": "c name",
              "uid": "c",
            },
          ]
    }
    onToggleCard={[Function]}
    selected={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CardList
        details={[Function]}
        label={[Function]}
        nodes={
                Array [
                        Object {
                          "age": "22",
                          "name": "a name",
                          "uid": "a",
                        },
                        Object {
                          "age": "88",
                          "name": "b name",
                          "uid": "b",
                        },
                        Object {
                          "age": "33",
                          "name": "c name",
                          "uid": "c",
                        },
                      ]
        }
        onToggleCard={[Function]}
        scrollTop={[Function]}
        selected={[Function]}
/>,
      "className": "scrollable",
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "details": [Function],
        "label": [Function],
        "nodes": Array [
          Object {
            "age": "22",
            "name": "a name",
            "uid": "a",
          },
          Object {
            "age": "88",
            "name": "b name",
            "uid": "b",
          },
          Object {
            "age": "33",
            "name": "c name",
            "uid": "c",
          },
        ],
        "onToggleCard": [Function],
        "scrollTop": [Function],
        "selected": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CardList
          details={[Function]}
          label={[Function]}
          nodes={
                    Array [
                              Object {
                                "age": "22",
                                "name": "a name",
                                "uid": "a",
                              },
                              Object {
                                "age": "88",
                                "name": "b name",
                                "uid": "b",
                              },
                              Object {
                                "age": "33",
                                "name": "c name",
                                "uid": "c",
                              },
                            ]
          }
          onToggleCard={[Function]}
          scrollTop={[Function]}
          selected={[Function]}
/>,
        "className": "scrollable",
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "details": [Function],
          "label": [Function],
          "nodes": Array [
            Object {
              "age": "22",
              "name": "a name",
              "uid": "a",
            },
            Object {
              "age": "88",
              "name": "b name",
              "uid": "b",
            },
            Object {
              "age": "33",
              "name": "c name",
              "uid": "c",
            },
          ],
          "onToggleCard": [Function],
          "scrollTop": [Function],
          "selected": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
